### PR TITLE
Add option to enable email notifications

### PR DESCRIPTION
**Pull request by dbkr**
Requires https://github.com/matrix-org/matrix-react-sdk/pull/289 (or will error)

https://github.com/vector-im/vector-web/issues/108

### CODE REVIEW

File: src/components/views/settings/Notifications.js [hunk 1]
```diff
@@ -620,6 +625,41 @@ module.exports = React.createClass({
...
     },
 
+    emailNotificationsRow: function(address, label) {
+        return (<div className="mx_UserNotifSettings_tableRow">
+            <div className="mx_UserNotifSettings_inputCell">
+                <input id="enableEmailNotifications_{address}"
+                    ref="enableEmailNotifications_{address}"
+                    type="checkbox"
+                    checked={ UserSettingsStore.hasEmailPusher(this.state.pushers, address) }
+                    onChange={ (e) => {
```
Comment (ara4n) [inline src/components/views/settings/Notifications.js]: feels like a bit of a weird conflation to inline the onChange implementation into the UI layout here, given it's not entirely trivial, and it breaks the pattern with the rest of the app (as far as I know), where any onChange longer than a few lines gets its own component method.  I don't object /too/ much, but what was the rationale, and is there a precedent?

**Comment by ara4n** [comment]
lgtm other than one comment.

**Comment by dbkr** [comment]
ptal

**Comment by ara4n** [comment]
lgtm

**Pull request merged**